packages/sidecar/src/cli.ts

```diff
@@ -0,0 +1,233 @@
+`);
+  process.exit(0);
+};
+
+export async function main(filesToServe?: Record<string, Buffer>) {
+  let runServer = true;
+  let { cmd, cmdArgs, help, port, debug } = parseCLIArgs();
+  if (debug || process.env.SPOTLIGHT_DEBUG) {
+    enableDebugLogging(true);
+  }
+  let stdioMCP = false;
+
+  if (help) {
+    runServer = false;
+    printHelp();
+  }
+
+  let onEnvelope: ((envelope: ParsedEnvelope["envelope"]) => void) | undefined = undefined;
+
+  switch (cmd) {
+    case "help":
+      printHelp();
+      runServer = false;
+      break;
+    case "mcp":
+      stdioMCP = true;
+      break;
+    // @ts-expect-error ts7029 -- fallthrough is intentional
+    case "run": {
+      let shell = false;
+      if (cmdArgs.length === 0) {
+        // try package.json to find default dev command
+        try {
+          const scripts = JSON.parse(readFileSync("package.json", "utf-8")).scripts;
+          cmdArgs = [scripts.dev ?? scripts.develop ?? scripts.serve ?? scripts.start];
+          shell = true;
+        } catch {
+          // pass
+        }
+      }
+      if (cmdArgs.length === 0) {
+        // try to see if this is a docker-compose project
+        try {
+          // TODO: Check docker-compose.yml existence
+          // TODO: Check docker-compose.yaml existence
+          // TODO: Check `docker compose --version` existence
+          // TODO: Check `docker-compose --version` existence
+        } catch {
+          // pass
+        }
+      }
+      if (cmdArgs.length === 0) {
+        logger.error("Error: No command specified to run and could not infer the command automatically.");
+        process.exit(1);
+      }
+      const cmdStr = cmdArgs.join(" ");
+      logger.info(`Starting command: ${cmdStr}`);
+      // TODO: Run this _after_ the server is started
+      const runCmd = spawn(cmdArgs[0], cmdArgs.slice(1), {
+        cwd: process.cwd(),
+        shell,
+        windowsVerbatimArguments: true,
+        windowsHide: true,
+        // TODO: Consider using pipe and forwarding output as logs to Spotlight
+        stdio: "ignore",
+        env: {
+          ...process.env,
+          SENTRY_SPOTLIGHT: getSpotlightURL(port),
+          // This is not supported in all SDKs but worth adding
+          // for the ones that support it
+          SENTRY_TRACES_SAMPLE_RATE: "1",
+          //  Consider setting SENTRY_DSN to our hack server too?
+        },
+      });
+      runCmd.on("error", err => {
+        logger.error(`Failed to run ${cmdStr}:`);
+        logger.error(err);
+        process.exit(1);
+      });
+      runCmd.on("close", code => {
+        if (!code) {
+          logger.error(`${cmdStr} exited, terminating.`);
+        } else {
+          logger.error(`${cmdStr} exited with code ${code}.`);
+        }
+        process.exit(code);
+      });
+      const killRunCmd = () => {
+        runCmd.removeAllListeners();
+        runCmd.kill();
+      };
```

Comment:
This could cause problems on windows as it doesn't kill the child processes.

I may be terribly wrong but it will fail on windows if we run:
- `spotlight run npm run dev`  the npm process gets killed, but not the dev server

Reply:
Hmm, good point. I'll see what we can do about this.